cmd/bosun/sched/host.go

```diff
@@ -0,0 +1,809 @@
+	silences := s.Silenced()
+	// These are all fetched by metric since that is how we store it in redis
+	// so this makes for the fastest response
+	osNetBytesTags, err := s.Search.FilteredTagSets("os.net.bytes", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetVirtualBytesTags, err := s.Search.FilteredTagSets("os.net.virtual.bytes", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetBondBytesTags, err := s.Search.FilteredTagSets("os.net.bond.bytes", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetTunnelBytesTags, err := s.Search.FilteredTagSets("os.net.tunnel.bytes", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetOtherBytesTags, err := s.Search.FilteredTagSets("os.net.other.bytes", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetIfSpeedTags, err := s.Search.FilteredTagSets("os.net.ifspeed", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetVirtualIfSpeedTags, err := s.Search.FilteredTagSets("os.net.virtual.ifspeed", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetBondIfSpeedTags, err := s.Search.FilteredTagSets("os.net.bond.ifspeed", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetTunnelIfSpeedTags, err := s.Search.FilteredTagSets("os.net.tunnel.ifspeed", nil)
+	if err != nil {
+		return nil, err
+	}
+	osNetOtherIfSpeedTags, err := s.Search.FilteredTagSets("os.net.other.ifspeed", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwChassisTags, err := s.Search.FilteredTagSets("hw.chassis", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwPhysicalDiskTags, err := s.Search.FilteredTagSets("hw.storage.pdisk", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwVirtualDiskTags, err := s.Search.FilteredTagSets("hw.storage.vdisk", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwControllersTags, err := s.Search.FilteredTagSets("hw.storage.controller", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwBatteriesTags, err := s.Search.FilteredTagSets("hw.storage.battery", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwPowerSuppliesTags, err := s.Search.FilteredTagSets("hw.ps", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwTempsTags, err := s.Search.FilteredTagSets("hw.chassis.temps.reading", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwBoardPowerTags, err := s.Search.FilteredTagSets("hw.chassis.power.reading", nil)
+	if err != nil {
+		return nil, err
+	}
+	diskTags, err := s.Search.FilteredTagSets("os.disk.fs.space_total", nil)
+	if err != nil {
+		return nil, err
+	}
+	// Will make the assumption that the metric bosun.ping.timeout, resolved, and rtt
+	// all share the same tagset
+	icmpTimeOutTags, err := s.Search.FilteredTagSets("bosun.ping.timeout", nil)
+	if err != nil {
+		return nil, err
+	}
+	for name, host := range hosts {
+		host.Name = name
+		hostTagSet := opentsdb.TagSet{"host": host.Name}
+		hostMetadata, err := s.GetMetadata("", hostTagSet)
+		if err != nil {
+			slog.Error(err)
+		}
+		processHostIncidents(host, states, silences)
+		for _, ts := range icmpTimeOutTags {
```

Comment:
this is a lot of iteration. I each iteration over hosts, you iterate over all tags for a lot of metrics. Perhaps we can turn those arrays into maps by host before we enter the main loop.